citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AsyncCitationProcessor.java

```diff
@@ -237,6 +240,17 @@ private void storeCitation(ICitation citation) {
         }
 
         citationStore.save((Citation) citation);
+        //If the citation is a deleted metadata note and if it is the active metadata note, remove its link from parent item
```

Comment:
Suppose an item had a metadata _note_ _a_ which is deleted by the user. After deletion, we update the citation and remove the link to this _note._ Later, the user edits the item and a new metadata _note_ _b_ is created. The item will be updated with the link to this _note_. Here, _note_ _b_ is the active metadata note. Even after deletion, _note_ _a_ can be edited and citesphere will try to update it. It would still qualify as a deleted metadata note but since it was already removed, we don't have to remove the current link to _note_ _b_ from the item.

Reply:
comment (in the code) needs to be clearer